docs/samples/v1beta1/triton/fastertransformer/README.md

```diff
@@ -0,0 +1,230 @@
+
+> **Note**  
+> This example requires access to a KServe cluster with GPU support
+
+In this example we will demonstrate:
+- Building a custom Triton image that includes the FasterTransformer backend
+- Defining a custom `ClusterServingRuntime`
+- Exporting the [BLOOM 560m](https://huggingface.co/bigscience/bloom-560m) model for FasterTransformer
+- Preparing the model for loading in Triton
+- Setting up and building our KServe transformer
+- Defining and deploying our inference service
+- Running inference on our model
+
+## Building our custom Triton image
+
+Since the officially supported Triton images do not contain the FasterTransformer backend we will need to build our own custom image. The FasterTransformer backend repo [provides detailed instructions on building a custom image](https://github.com/triton-inference-server/fastertransformer_backend#prepare-docker-images). Build the image according to the instructions and push it to a container repository that is accessible by your KServe cluster.
+
+## Setting up a `ClusterServingRuntime`
+
+In order to allow our inference services to easily use our new image we will setup a custom cluster serving runtime. Apply the resource to your cluster as follows, making sure to replace the `image` field with the one you created in the previous step.
+
+
+```yaml
+apiVersion: serving.kserve.io/v1alpha1
+kind: ClusterServingRuntime
+metadata:
+  name: tritonserver-ft
+spec:
+  annotations:
+    prometheus.kserve.io/path: /metrics
+    prometheus.kserve.io/port: "8002"
+  containers:
+  - args:
+    - tritonserver
+    - --model-store=/mnt/models
+    - --grpc-port=9000
+    - --http-port=8080
+    - --allow-grpc=true
+    - --allow-http=true
+    image: <my-custom-triton-image>
+    name: kserve-container
+    resources:
+      limits:
+        cpu: "1"
+        memory: 2Gi
+      requests:
+        cpu: "1"
+        memory: 2Gi
+  protocolVersions:
+  - v2
+  - grpc-v2
+  supportedModelFormats:
+  - name: triton
+    version: "2"
+```
+
+## Exporting the BLOOM 560m model
+
+The FasterTransformer repo provides a script for converting the HuggingFace BLOOM model to FasterTransformer format. The following directions are based on those outlined in the [FasterTransformer backend repo](https://github.com/triton-inference-server/fastertransformer_backend/blob/main/docs/gpt_guide.md#run-bloom).
+
+1. Create a working directory
+```sh
+mkdir ~/triton-bloom
+cd ~/triton-bloom
+```
+
+2. Clone the HuggingFace BLOOM 560m repository
+
+```sh
+# Make sure you have git-lfs installed (https://git-lfs.com)
+git lfs install
+git clone https://huggingface.co/bigscience/bloom-560m
+```
+
+3. Fetch the conversion script
+
+```sh
+curl -LO https://raw.githubusercontent.com/NVIDIA/FasterTransformer/main/examples/pytorch/gpt/utils/huggingface_bloom_convert.py
+```
+
+4. Convert the HuggingFace model to FT format
+
+```sh
+python3 huggingface_bloom_convert.py -o bloom -i ./bloom-560m/ -tp 1
```

Comment:
We can add some explanation here for what tp means here and how they can shard the model if it is too big